File: agriTech/src/main/java/com/example/agriTech/dto/plantuser/PlantUserDTO.java
package com.example.agriTech.dto.plantuser;

import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.Data;
@Data
public class PlantUserDTO {
    private Long id; // ID của plants_users
    private String customName;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime lastWatered;
    private Boolean status;
    // Thuộc tính từ bảng Plant
    private Long plantId;
    private String plantName;
    private String imageUrl;
    private String fertilizerInfo;

    // Thuộc tính từ bảng Account/User
    private String FullName;
    private Long userId;
    private String deviceCode;

    
}

File: agriTech(DA3K)/app/src/main/java/com/example/agritechda3k/api/dto/PlantUserDTO.kt
package com.example.agritechda3k.api.dto

data class PlantUserDTO(
    val id: Long,
    val customName: String?,
    val lastWatered: String?, // Bên Spring gọt về String rồi thì ở đây để String
    val status: Boolean?,
    val plantId: Long,
    val plantName: String?,
    val imageUrl: String?,
    val fertilizerInfo: String?,
    val fullName: String?, // Nếu bên Spring ông đặt là FullName thì ở đây phải là FullName
    val userId: Long,
    val deviceCode:String
)
